src/cov.jl

```diff
@@ -44,80 +37,73 @@ that the data are centered and hence there's no need to subtract the mean.
 When `vardim = 1`, the variables are considered columns with observations in rows;
 when `vardim = 2`, variables are in rows with observations in columns.
 """
-function scattermat end
+scattermatm(x::DenseMatrix, mean, vardim::Int=1) =
+    scattermat_zm(x .- mean, vardim)
 
+scattermatm(x::DenseMatrix, mean, wv::AbstractWeights, vardim::Int=1) =
+    scattermat_zm(x .- mean, wv, vardim)
 
-"""
-    cov(X, wv::WeightVec; mean=nothing, vardim=1)
+scattermat(x::DenseMatrix, vardim::Int=1) =
+    scattermatm(x, Base.mean(x, vardim), vardim)
 
-Compute the weighted covariance matrix. By default, the covariance
-matrix is normalized by the sum of the weights. That is, `cov(X, wv)`
-is equivalent to `scattermat(X, wv) / sum(wv)`.
-"""
-cov
+scattermat(x::DenseMatrix, wv::AbstractWeights, vardim::Int=1) =
+    scattermatm(x, Base.mean(x, wv, vardim), wv, vardim)
 
+scattermat_zm(x::DenseMatrix, vardim::Int) = Base.unscaled_covzm(x, vardim)
 
-"""
-    mean_and_cov(x, [wv::WeightVec]; vardim=1) -> (mean, cov)
+scattermat_zm(x::DenseMatrix, wv::AbstractWeights, vardim::Int) =
+    _symmetrize!(Base.unscaled_covzm(x, _scalevars(x, values(wv), vardim), vardim))
 
-Return the mean and covariance matrix as a tuple. A weighting
-vector `wv` can be specified. `vardim` that designates whether
-the variables are columns in the matrix (`1`) or rows (`2`).
 """
-function mean_and_cov end
-
+    cov(X, wv::AbstractWeights, [vardim, corrected])
 
-@static if VERSION < v"0.5.0-dev+679"
-    function scattermat(x::DenseMatrix; mean=nothing, vardim::Int=1)
-        mean == 0 ? scattermat_zm(x, vardim) :
-        mean == nothing ? scattermat_zm(x .- Base.mean(x, vardim), vardim) :
-        scattermat_zm(x .- mean, vardim)
-    end
+Compute the weighted covariance matrix. Similar to `var` and `std` the biased covariance
+matrix (`corrected=false`) can be computed by multiplying `scattermat(X, wv)` by
+``\frac{1}{\sum{w}}`` to normalize. However, the unbiased covariance matrix
+(`corrected=true`) is dependent on the type of weights used:
 
-    function scattermat(x::DenseMatrix, wv::WeightVec; mean=nothing, vardim::Int=1)
-        mean == 0 ? scattermat_zm(x, wv, vardim) :
-        mean == nothing ? scattermat_zm(x .- Base.mean(x, wv, vardim), wv, vardim) :
-        scattermat_zm(x .- mean, wv, vardim)
-    end
+* AnalyticWeights: ``\\frac{1}{\sum w - \sum {w^2} / \sum{w}^2}``
```

Comment:
I guess it should be ``\\frac{1}{\sum w - \sum {w^2} / \sum w}``, which wouldn't be a problem if `w` was prenormalized, but yeah.
```julia
julia> cvar1(w) = sum(w) * (1 - sum(normalize(w, 1) .^ 2))
cvar1 (generic function with 1 method)

julia> cvar2(w) = sum(w) - (sum(w .^2) / sum(w))
cvar2 (generic function with 1 method)

julia> w = rand(10)
10-element Array{Float64,1}:
 0.608948
 0.980859
 0.656496
 0.994447
 0.912615
 0.654867
 0.0928164
 0.308836
 0.782151
 0.952136

julia> cvar1(w)
6.132433099237098

julia> cvar2(w)
6.132433099237098
```